cylc/flow/scheduler.py

```diff
@@ -1123,7 +1122,7 @@ def load_flow_file(self, is_reload=False):
             self.workflow, f"{time_str}-{load_type}.cylc")
         with open(file_name, "wb") as handle:
             handle.write(b"# cylc-version: %s\n" % CYLC_VERSION.encode())
-            printcfg(self.config.cfg, none_str=None, handle=handle)
+            self.config.pcfg.idump(sparse=True, handle=handle)
```

Comment:
Might want to open this file in text mode (rather than bytes mode) to avoid having to trickle `if isinstance(string, bytes)` like logic down the call stack.

Reply:
Yes